### ISSUE TITLE
[x264] build failure

### ISSUE THREAD

**Issue by HppZ** [microsoft/vcpkg#15786]
**Host Environment**
- OS: [Windows 19041]
- Compiler: vs2019 16.8.4

building x264 with 'vcpkg install x264 --triplet=x64-uwp' does not work.

```pswh
PS C:\Users\username\Documents\username\opensource\vcpkg> vcpkg install x264 --triplet=x64-uwp 
Computing installation plan... 
The following packages will be built and installed: x264[core]:x64-uwp -> 161-0d754ec36013fee82978496cd56fbd48824910b3#15 Detecting compiler hash for triplet x64-uwp... 
Could not locate cached archive: C:\Users\username\AppData\Local\vcpkg\archives\06\0661a1f5d0e56d348ea6638e281d4c5d772cf081.zip 
Starting package 1/1: x264:x64-uwp Building package x264[core]:x64-uwp... 
-- Using cached C:/Users/username/Documents/username/opensource/vcpkg/downloads/mirror-x264-0d754ec36013fee82978496cd56fbd48824910b3.tar.gz 
-- Cleaning sources at C:/Users/username/Documents/username/opensource/vcpkg/buildtrees/x264/src/48824910b3-cab15fc14a.clean. Use --editable to skip cleaning for the packages you specify. 
-- Extracting source C:/Users/username/Documents/username/opensource/vcpkg/downloads/mirror-x264-0d754ec36013fee82978496cd56fbd48824910b3.tar.gz 
-- Using source at C:/Users/username/Documents/username/opensource/vcpkg/buildtrees/x264/src/48824910b3-cab15fc14a.clean 
-- Configuring x64-uwp 
-- Using msys root at C:/Users/username/Documents/username/opensource/vcpkg/downloads/tools/msys2/23ab37a7454cafa4 
-- Using msys root at C:/Users/username/Documents/username/opensource/vcpkg/downloads/tools/msys2/c809757c94447846
-- Configuring x64-uwp-dbg 
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:108 (message): 
Command failed: C:/Users/username/Documents/username/opensource/vcpkg/downloads/tools/msys2/23ab37a7454cafa4/usr/bin/bash.exe --noprofile --norc --debug -c "V=1 CPP='cl.exe -E' CC='cl.exe' CC_FOR_BUILD='cl.exe' CXX='cl.exe' AR='lib.exe' LD='link.exe -verbose' RANLIB=':' STRIP=':' NM='dumpbin.exe -symbols -headers' DLLTOOL='link.exe -verbose -dll' CCAS=':' AS='C:/Users/username/Documents/username/opensource/vcpkg/downloads/tools/nasm/nasm-2.15.05/nasm.exe' ./../src/48824910b3-cab15fc14a.clean/configure --build=x86_64-pc-mingw32 --host=x86_64-pc-mingw32 --extra-cflags=-DWINAPI_FAMILY=WINAPI_FAMILY_APP --extra-cflags=-D_WIN32_WINNT=0x0A00 --extra-ldflags=-APPCONTAINER --extra-ldflags=WindowsApp.lib --disable-asm --enable-strip --disable-lavf --disable-swscale --disable-avs --disable-ffms --disable-gpac --disable-lsmash --enable-debug ac_cv_prog_ac_ct_STRIP=: gl_cv_double_slash_root=yes ac_cv_func_memmove=yes --disable-silent-rules --verbose --enable-shared --disable-static --prefix=/C/Users/username/Documents/username/opensource/vcpkg/installed/x64-uwp/debug" 
Working Directory: C:/Users/username/Documents/username/opensource/vcpkg/buildtrees/x264/x64-uwp-dbg 
Error code: 1 
See logs for more information: 
C:\Users\username\Documents\username\opensource\vcpkg\buildtrees\x264\config-x64-uwp-dbg-out.log
Call Stack (most recent call first):
scripts/cmake/vcpkg_configure_make.cmake:679 (vcpkg_execute_required_process)
ports/x264/portfile.cmake:35 (vcpkg_configure_make)
scripts/ports.cmake:136 (include)

Error: Building package x264:x64-uwp failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with .\vcpkg update, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
Package: x264:x64-uwp
Vcpkg version: 2020.11.12--unknownhash

Additionally, attach any relevant sections from the log files above.
```

**Comment by JackBoosY** [comment]
Please provide failure log _C:\Users\username\Documents\username\opensource\vcpkg\buildtrees\x264\config-x64-uwp-dbg-out.log_.

Thanks.

**Comment by HppZ** [comment]
no x264.exe generated.

**Comment by JackBoosY** [comment]
@HppZ That will be fixed in #15769.

**Comment by JackBoosY** [comment]
Duplicated to #15744.
